evproc/processor.py

```diff
@@ -308,11 +311,16 @@ def process(self, ev):
                   ``StopProcessing`` exception raised by a processor.
         """
 
-        # Walk through all the processors and process the event
-        for proc in self._get_procs(ev):
-            try:
-                proc(ev)
-            except exc.StopProcessing as ex:
-                return ex.retval
+        event_deque = collections.deque([ev])
+        while event_deque:
+            ev = event_deque.popleft()
+            # Walk through all the processors and process the event
+            for proc in self._get_procs(ev):
+                try:
+                    additional_events = proc(ev)
+                except exc.StopProcessing as ex:
+                    return ex.retval
```

Comment:
Hmmm…I like the split on StopProcessing approach.  It could even be done using the existing StopProcessing; just add an additional argument to signal which behavior to use, and give it a reasonable default.